### PR TITLE
[SPARK-33690][SQL][FOLLOWUP] Escape further meta-characters in showString

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

This is a followup PR for SPARK-33690 (#30647) .
In addition to the original PR, this PR intends to escape the following meta-characters in `Dataset#showString`.

  * `\r` (carrige ret)
  * `\f` (form feed)
  * `\b` (backspace)
  * `\u000B` (vertical tab)
  * `\u0007` (bell)

### Why are the changes needed?

To avoid breaking the layout of `Dataset#showString`.
`\u0007` does not break the layout of `Dataset#showString` but it's noisy (beeps for each row) so it should be also escaped.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Modified the existing tests.
I also build the documents and check the generated html for `sql-migration-guide.md`.

### CODE REVIEW

**Comment by sarutak** [comment]
retest this please.

**Review by srowen — APPROVED** [review]

File: docs/sql-migration-guide.md [hunk 1]
```diff
@@ -26,7 +26,14 @@ license: |
 
   - In Spark 3.2, `spark.sql.adaptive.enabled` is enabled by default. To restore the behavior before Spark 3.2, you can set `spark.sql.adaptive.enabled` to `false`.
 
-  - In Spark 3.2, the meta-characters `\n` and `\t` are escaped in the `show()` action. In Spark 3.1 or earlier, the two metacharacters are output as it is.
+  - In Spark 3.2, the following meta-characters are escaped in the `show()` action. In Spark 3.1 or earlier, the two metacharacters are output as it is.
```
Comment (dongjoon-hyun) [inline docs/sql-migration-guide.md:29]: In the second sentence, `two` -> `following`?

**Review by dongjoon-hyun — COMMENTED** [review]
cc @maropu

File: docs/sql-migration-guide.md (hunk 1 above)
Comment (sarutak) [reply docs/sql-migration-guide.md:29]: I missed. Thanks.

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 2]
```diff
@@ -310,7 +310,14 @@ class Dataset[T] private[sql](
...
             // Escapes meta-characters not to break the `showString` format
-            cell.toString.replaceAll("\n", "\\\\n").replaceAll("\t", "\\\\t")
+            cell.toString
+              .replaceAll("\n", "\\\\n")
+              .replaceAll("\r", "\\\\r")
+              .replaceAll("\t", "\\\\t")
+              .replaceAll("\f", "\\\\f")
+              .replaceAll("\b", "\\\\b")
+              .replaceAll("\u000B", "\\\\v")
+              .replaceAll("\u0007", "\\\\a")
```
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala:320]: It seems `fieldNames` (L306) has the same issue, so could you fix it in this PR, too?
```
scala> spark.range(1).selectExpr("'aaa\nbbb'").show()
+--------+                                                                      
| aaa
bbb|
+--------+
|aaa\nbbb|
+--------+
```

Comment (sarutak) [reply sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala:320]: Sure, will do.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**